From 2364113bcbeda36d1b0a12d5612faa8abca0457d Mon Sep 17 00:00:00 2001
From: Omri Itach <omrii@marvell.com>
Date: Sun, 2 Apr 2017 14:55:52 +0300
Subject: [PATCH 0932/1345] clk: ap806: add support for CPU frequnecy clock at
 2GHz

commit  21420f812cf2b9e146161fd70ffe4c6d33b41faa from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

added 2 mising Sample at reset entries to AP806 syscon driver:
0x0: CPU@2000Mhz , RCLK/DDR@1200 Mhz , DCLK@600Mhz, PIDI@1000Mhz
0x1: CPU@2000Mhz , RCLK/DDR@1050 Mhz , DCLK@525Mhz, PIDI@1000Mhz

This frequency addition also fixes issues in CPUfreq driver with CPU@2Ghz.

Change-Id: Id46a50502529b3d467ff6773f249808a3effee57
Signed-off-by: Omri Itach <omrii@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/38342
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Hanna Hawa <hannah@marvell.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/clk/mvebu/ap806-system-controller.c |    6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/clk/mvebu/ap806-system-controller.c b/drivers/clk/mvebu/ap806-system-controller.c
index dc34d4b..ee82a9b 100644
--- a/drivers/clk/mvebu/ap806-system-controller.c
+++ b/drivers/clk/mvebu/ap806-system-controller.c
@@ -99,6 +99,12 @@ static int ap806_syscon_clk_probe(struct platform_device *pdev)
 
 	/* Get DCLK frequency */
 	switch (freq_mode) {
+	case 0x0:
+		dclk_freq = 600;
+		break;
+	case 0x1:
+		dclk_freq = 525;
+		break;
 	case 0x4:
 	case 0x10:
 	case 0x14:
-- 
1.7.9.5

